.github/workflows/terragrunt-apply-production.yml

```diff
@@ -189,7 +190,7 @@ jobs:
         run: terragrunt apply --terragrunt-non-interactive -auto-approve
 
   update-lambda-function-image:
-    needs: [detect-lambda-changes, build-tag-push-lambda-images, terragrunt-apply-all-modules]
+    needs: [detect-lambda-changes, terragrunt-apply-all-modules]
```

Comment:
Removed the `build-tag-push-lambda-images` dependency as the `terragrunt-apply-all-modules` job already waits for the `build-tag-push-lambda-images` to be completed